src/autopas/AutoPasDecl.h

```diff
@@ -130,25 +130,89 @@ class AutoPas {
    */
   [[nodiscard]] std::vector<Particle> updateContainer();
 
+  /**
+   * Reserve memory for a given number of particles in the container and logic layers.
+   * This function assumes a uniform distribution of particles throughout the domain.
+   * For example, this means that in a LinkedCells Container in each cell vector.reserve(numParticles/numCells) is
+   * called.
+   * @note This functions will create an estimate for the number of halo particles.
+   * @param numParticles No buffer factor is applied. It is probably wise to slighly over-reserve to account for
+   * imbalance or particle movement.
+   */
+  void reserve(size_t numParticles);
+
+  /**
+   * Reserve memory for a given number of particles in the container and logic layers
+   * (e.g. LogicHandler::_particleBuffer).
+   * This function assumes a uniform distribution of particles throughout the domain.
+   * For example, this means that in a LinkedCells Container in each cell vector.reserve(numParticles/numCells) is
+   * called.
+   * @param numParticles
+   * @param numHaloParticles
+   */
+  void reserve(size_t numParticles, size_t numHaloParticles);
+
   /**
    * Adds a particle to the container.
    * This is only allowed if the neighbor lists are not valid.
    * @param p Reference to the particle to be added
    * @note An exception is thrown if the particle is added and it is not inside of the owned domain (defined by
    * boxmin and boxmax) of the container.
-   * @note This function is NOT thread-safe.
+   * @note This function is NOT thread-safe if the container is Octree.
    */
   void addParticle(const Particle &p);
 
+  /**
+   * Adds all particles from the collection to the container.
+   * @note This function uses addParticlesIf().
```

Comment:
I don't see where this function uses `addParticlesIf()` 
```suggestion
   * @note This function uses addParticle().
```